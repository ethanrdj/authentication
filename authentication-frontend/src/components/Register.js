import React from "react";
import { Link } from "react-router-dom";
import Alert from "./Alert";
import PropTypes from "prop-types";

const Register = ({
  handleChange,
  handleSubmitRegister,
  alertMessage,
  setAlertMessage,
}) => {
  return (
    <div className="Register">
      <h1 className="header">Register</h1>
      <form className="form" onSubmit={handleSubmitRegister}>
        <label className="form-labels" htmlFor="register-email">
          Email
        </label>
        <input
          className="form-input"
          id="register-email"
          type="email"
          name="email"
          required
          onChange={handleChange}
        />
        <label className="form-labels" htmlFor="register-username">
          Username
        </label>
        <input
          className="form-input"
          id="register-username"
          type="text"
          name="username"
          required
          onChange={handleChange}
        />
        <label className="form-labels" htmlFor="register-password">
          Password
        </label>
        <input
          className="form-input"
          id="register-password"
          type="password"
          name="password"
          required
          onChange={handleChange}
        />
        <label className="form-labels" htmlFor="confirm-password">
          Confirm Password
        </label>
        <input
          className="form-input"
          id="confirm-password"
          type="password"
          name="confirmPassword"
          required
          onChange={handleChange}
        />

        {alertMessage && <Alert alertMessage={alertMessage} />}

        <button className="form-button">Register and Login</button>

        <Link className="link-to" to="/" onClick={() => setAlertMessage("")}>
          Already have an account? Login here!
        </Link>
      </form>
    </div>
  );
};

Register.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmitRegister: PropTypes.func.isRequired,
  alertMessage: PropTypes.string,
  setAlertMessage: PropTypes.func,
};

export default Register;
