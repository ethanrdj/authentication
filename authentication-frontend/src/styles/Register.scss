.register-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.register-input {
  width: 60%;
}
