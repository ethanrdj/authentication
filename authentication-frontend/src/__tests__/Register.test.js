import { render } from "@testing-library/react";
import Register from "../components/Register";
import { BrowserRouter } from "react-router-dom";

jest.mock("../components/Alert", () => () => <div data-testid="alert-id" />);

describe("Register", () => {
  it("renders the component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Register
          handleChange={jest.fn()}
          handleSubmitRegister={jest.fn()}
          alertMessage="Test alert message"
          setAlertMessage={jest.fn()}
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct elements on the page", () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Register
          handleChange={jest.fn()}
          handleSubmitRegister={jest.fn()}
          alertMessage="Test alert message"
          setAlertMessage={jest.fn()}
        />
      </BrowserRouter>
    );

    expect(getByTestId("reg-email-id")).toHaveTextContent("Email");
    expect(getByTestId("reg-username-id")).toHaveTextContent("Username");
    expect(getByTestId("reg-password-id")).toHaveTextContent("Password");
    expect(getByTestId("reg-confirm-id")).toHaveTextContent("Confirm Password");
  });

  it("renders the alert message", () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Register
          handleChange={jest.fn()}
          handleSubmitRegister={jest.fn()}
          alertMessage="Test alert message"
          setAlertMessage={jest.fn()}
        />
      </BrowserRouter>
    );

    expect(getByTestId("alert-id")).toBeInTheDocument();
  });
});
