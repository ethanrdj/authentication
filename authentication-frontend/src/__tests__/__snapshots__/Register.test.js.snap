// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders the component 1`] = `
<DocumentFragment>
  <div
    class="Register"
  >
    <h1
      class="header"
    >
      Register
    </h1>
    <form
      class="form"
    >
      <label
        class="form-labels"
        data-testid="reg-email-id"
        for="register-email"
      >
        Email
      </label>
      <input
        class="form-input"
        id="register-email"
        name="email"
        required=""
        type="email"
      />
      <label
        class="form-labels"
        data-testid="reg-username-id"
        for="register-username"
      >
        Username
      </label>
      <input
        class="form-input"
        id="register-username"
        name="username"
        required=""
        type="text"
      />
      <label
        class="form-labels"
        data-testid="reg-password-id"
        for="register-password"
      >
        Password
      </label>
      <input
        class="form-input"
        id="register-password"
        name="password"
        required=""
        type="password"
      />
      <label
        class="form-labels"
        data-testid="reg-confirm-id"
        for="confirm-password"
      >
        Confirm Password
      </label>
      <input
        class="form-input"
        id="confirm-password"
        name="confirmPassword"
        required=""
        type="password"
      />
      <div
        data-testid="alert-id"
      />
      <button
        class="form-button"
      >
        Register and Login
      </button>
      <a
        class="link-to"
        href="/"
      >
        Already have an account? Login here!
      </a>
    </form>
  </div>
</DocumentFragment>
`;
